Warning on rtttl during compiling: const defined but not used

### The problem

During compilation (installing) on my ESPHome-device i see a warning

```
Compiling .pioenvs/wemosd1-r1s/src/esphome/components/rtttl/rtttl.cpp.o
src/esphome/components/rtttl/rtttl.cpp:378:25: warning: 'const esphome::LogString* esphome::rtttl::state_to_string(esphome::rtttl::State)' defined but not used [-Wunused-function]
  378 | static const LogString *state_to_string(State state) {
      |                         ^~~~~~~~~~~~~~~

```


### Which version of ESPHome has the issue?

2025.9.1 

### What type of installation are you using?

Home Assistant Add-on

### What platform are you using?

ESP8266

### Component causing the issue

rtttl

### YAML Config

```yaml
esphome:
  name: "wemosd1-r1s"

  # do something on boot
  # default priority = 600 => most sensors are set up. Before 250 = Wifi is initialized
  on_boot:
    then:
      - switch.turn_on: internal_pin    # turn on = inverted pin = on => led=on
      - delay: 500ms 
      - switch.turn_off: internal_pin   # turn off = inverted pin = off => led=off
      - delay: 500ms 
      - switch.turn_on: internal_pin    # turn on = inverted pin = on => led=on
      - delay: 500ms 
      - switch.turn_off: internal_pin   # turn off = inverted pin = off => led=off
      - delay: 500ms 

substitutions:
  devicename: roomsensor1

# Connected sensors to this device
#  - internal led  on GPIO2   = D4
#  - CO2-sensor    on UART: RX/GPIO3 + TX/GPIO1
#  - RGBled PL9823 on GPIO15  = D8
#  - Buzzer        on GPIO13  = D7
#  - HDC1080       on I²C: SCL/GPIO5/D1 + SDA/GPIO4/D2 + 3V3 + GND
#  - PIR           on GPIO12  = D6
#  - LDR resistor  on A0
# ESPHome settings
#  - (default) logging
#  - (default) HomeAssistant API
#  - (default) OTA
#  - 3x Wifi networks from secret
#  - (default) fallback hotspot
#  - info to HA: wifi-signal, ip-address, connectedSSID, Mac wifi, latest SSID scan
#  - info to HA: uptime, status, restart button

esp8266:
  board: d1_mini_pro
  # board: esp01_1m

# Enable logging
# Got warning on using CO2-MH-Z19C-sensor: You're using the same serial port 
# for logging and the UART component. Please disable logging over the serial 
# port by setting logger->baud_rate to 0.
logger:
  baud_rate: 0

ota:
  platform: esphome

# Software-PWM on D2 for passive buzzer
# pin -> connected to GND
# pin -> connected to D7 = GPIO13
output:
  - platform: esp8266_pwm
    pin: D7
    id: rtttl_out

# RTTTL = Ring Tone Text Transfer Language
# see https://esphome.io/components/rtttl.html
rtttl:
  output: rtttl_out
  on_finished_playback:
    - logger.log: 'Song ended!'

# Generate API (for HomeAssistant) to play a 'song'/melody/tone
api:
  services:
    - service: play_rtttl
      variables:
        song_str: string
      then:
        - rtttl.play:
            rtttl: !lambda 'return song_str;'

# from https://esphome.io/components/sensor/mhz19.html
# As the communication with the MH-Z19 is done using UART, you need to have an
# UART bus in your configuration with the rx_pin connected to the TX pin of the
# MH-Z19 and the tx_pin connected to the RX Pin of the MH-Z19 (it’s switched 
# because the TX/RX labels are from the perspective of the MH-Z19).
# Additionally, you need to set the baud rate to 9600.
uart:
  rx_pin: GPIO3
  tx_pin: GPIO1
  baud_rate: 9600

# I2C bus to handle HDC1080 temp sensor
i2c:
  sda: 4      # GPIO4 / D2
  scl: 5      # GPIO5 / D1
  scan: true  # default=true, optional
  # id: bus_a   # optional

wifi:
  # reversed order to try no 2 = bedroom first
  networks:
  - ssid: !secret wifi_ssid2            # IoT
    password: !secret wifi_password2
  - ssid: !secret wifi_ssid4            # EcoZonnewoning
    password: !secret wifi_password4
  - ssid: !secret wifi_ssid3            # slaapk
    password: !secret wifi_password3
  - ssid: !secret wifi_ssid             # Zonnewoning
    password: !secret wifi_password

  # Enable fallback hotspot (captive portal) in case wifi connection fails
  ap:
    ssid: "Esphome-RS1-04F9C6"
    password: !secret ap_password_for_all

captive_portal:

light:
  # RGB led PL9823
  # pinout van bol naar platte kant (5mm)
  # pin 1 = (short) Data In  = DI   -> connected to D0 / GPIO16
  # pin 2 = (short) V+       = 5V   -> connected to 5V
  # pin 3 = (long)  GND      = GND  -> connected to GND
  # pin 4 = (long)  Data Out = DO   -> not connected
  - platform: neopixelbus
    name: ${devicename} RGBled
    num_leds: 1
    variant: WS2811
    pin: GPIO15  #GPIO15 / D8  
    type: RGB

switch:
# internal led is available
- platform: gpio
  pin: 
    number: 2 #GPIO2 / D4 is internal_led_pin
    inverted: true     # output low = pin off = led ON
  id: internal_pin
  restore_mode: ALWAYS_OFF   # start low = pin on = led OFF
# switch functionality, LED is on if switch is turned off !
- platform: template
  name: ${devicename} internal LED
  optimistic: yes
  id: iled
  turn_on_action:
  - switch.turn_on: internal_pin    # turn on = inverted pin = on => led=on
  turn_off_action:
  - switch.turn_off: internal_pin   # turn off = inverted pin = off => led=off
   
sensor:
  # CO2 sensor MH-Z19C
  - platform: mhz19
    co2:
      name: ${devicename} CO2 Value
    temperature:
      name: ${devicename} CO2 Temperature
    update_interval: 60s
    automatic_baseline_calibration: false

  # Temp/Hum sensor HDC1080 on I2C
  - platform: hdc1080
    temperature:
      name: ${devicename} Temperature
      filters:
        - lambda: !lambda |-
            if (x < 1) return {};
            return x;
    humidity:
      name: ${devicename} Humidity
    update_interval: 60s

  # Light resistance sensor LDR, calculated from ADC
  # T------ 3V3
  # |
  # R = 20kOhm
  # |
  # E------ pin A0
  # |
  # LDR
  # |
  # L------ GND
  - platform: resistance
    sensor: R1S_a0_sensor
    configuration: DOWNSTREAM
    resistor: 20.0kOhm
    name: ${devicename} LDR
    
  # Analog input from LDR
  - platform: adc
    id: R1S_a0_sensor
    pin: A0
    update_interval: 10s
    name: ${devicename} A0 reading
    
    
  # ESPHome Wifi information
  - platform: wifi_signal
    name: ${devicename} Wifi signal
    update_interval: 60s

  # ESPHome uptime
  - platform: uptime
    name: ${devicename} Uptime seconds
  - platform: uptime
    name: ${devicename} Uptime readable
    id: uptime_sensor
    update_interval: 60s
    on_raw_value:
      then:
        - text_sensor.template.publish:
            id: uptime_human
            state: !lambda |-
              int seconds = round(id(uptime_sensor).raw_state);
              int days = seconds / (24 * 3600);
              seconds = seconds % (24 * 3600);
              int hours = seconds / 3600;
              seconds = seconds % 3600;
              int minutes = seconds /  60;
              seconds = seconds % 60;
              return (
                (days ? to_string(days) + "d " : "") +
                (hours ? to_string(hours) + "h " : "") +
                (minutes ? to_string(minutes) + "m " : "") +
                (to_string(seconds) + "s")
              ).c_str();
   
binary_sensor:
  # PIR motion sensor AM312
  # PIN VIN / + -> connected to 3V3
  # PIN OUT     -> connected to D6 = GPIO12
  # PIN GND / - -> connected to GND
  - platform: gpio
    pin: D6   # GPIO12 = D6
    name: ${devicename} Motion
    device_class: motion

  # Is the device connected or not?
  - platform: status
    name: ${devicename} Status

text_sensor:
  # ESPHome version
  - platform: version
    name: ${devicename} ESPHome version
    hide_timestamp: true
    
  # ESPHome Wifi information
  - platform: wifi_info
    ip_address:
      name: ${devicename} IP Address
    ssid:
      name: ${devicename} Connected SSID
    mac_address:
      name: ${devicename} MAC Wifi
    scan_results:
      name: ${devicename} Latest Scan

  # ESPHome uptime readable text in stead of seconds
  - platform: template
    name: ${devicename} Uptime Human Readable
    id: uptime_human
    icon: mdi:clock-start
```

### Anything in the logs that might be useful for us?

```txt

```

### Additional information

I have seen the warning before, so it is not related to changes in ESPHome v2025.9.1
And also on other of my sensors (with an rtttl buzzer)

--- a/esphome/components/rtttl/rtttl.cpp
+++ b/esphome/components/rtttl/rtttl.cpp
@@ -374,7 +374,7 @@ void Rtttl::loop() {
   this->last_note_ = millis();
 }
 
-#if ESPHOME_LOG_LEVEL >= ESPHOME_LOG_LEVEL_DEBUG
+#if ESPHOME_LOG_LEVEL >= ESPHOME_LOG_LEVEL_VERBOSE
 static const LogString *state_to_string(State state) {
   switch (state) {
     case STATE_STOPPED:

